substrate-node/pallets/pallet-smart-contract/src/cost.rs

```diff
@@ -287,22 +319,19 @@ pub fn calculate_cost_in_tft_from_musd<T: Config>(
 ) -> Result<u64, DispatchErrorWithPostInfo> {
     let avg_tft_price = pallet_tft_price::AverageTftPrice::<T>::get();
 
-    // Guaranty tft price will never be lower than min tft price
+    // Guarantee tft price will never be lower than min tft price
     let min_tft_price = pallet_tft_price::MinTftPrice::<T>::get();
     let mut tft_price = avg_tft_price.max(min_tft_price);
 
-    // Guaranty tft price will never be higher than max tft price
+    // Guarantee tft price will never be higher than max tft price
     let max_tft_price = pallet_tft_price::MaxTftPrice::<T>::get();
     tft_price = tft_price.min(max_tft_price);
 
-    // TFT Price is in musd
-    let tft_price_musd = U64F64::from_num(tft_price);
-
-    // Cost is expressed in units USD, divide by 10000 to get the price in musd
-    let total_cost_musd = U64F64::from_num(total_cost) / 10000;
+    // TFT Price is in musd so lets convert to units usd
+    let tft_price_units_usd = U64F64::from_num(tft_price) * 10000;
```

Comment:
Detail, but  would put here
~~~rust
let tft_price_units_usd = tft_price * 10000;
~~~
and then below
~~~rust
let total_cost_tft = U64F64::from_num(total_cost) / U64F64::from_num(tft_price_units_usd);
~~~
for readability